projects/packages/publicize/src/class-rest-controller.php

```diff
@@ -158,11 +158,61 @@ public function register_rest_routes() {
 			array(
 				'methods'             => WP_REST_Server::DELETABLE,
 				'callback'            => array( $this, 'delete_publicize_connection' ),
-				'permission_callback' => array( $this, 'require_author_privilege_callback' ),
+				'permission_callback' => array( $this, 'manage_connection_permission_check' ),
 			)
 		);
 	}
 
+	/**
+	 * Manage connection permission check
+	 *
+	 * @param WP_REST_Request $request The request object, which includes the parameters.
+	 *
+	 * @return bool True if the user can manage the connection, false otherwise.
+	 */
+	public function manage_connection_permission_check( WP_REST_Request $request ) {
+
+		if ( current_user_can( 'edit_others_posts' ) ) {
+			return true;
+		}
+
+		/**
+		 * Publicize instance.
+		 *
+		 * @var Publicize $publicize Publicize instance.
+		 */
+		global $publicize;
+
+		$connection = $publicize->get_connection_for_user( $request->get_param( 'connection_id' ) );
+
+		$owns_connection = get_current_user_id() === (int) $connection['user_id'];
```

Comment:
Should we check if isset( $connection['user_id'] ) before accessing it?

Reply:
I think that is always present but yeah, better to be safe than sorry.

Reply:
Done in https://github.com/Automattic/jetpack/pull/38187/commits/0d4dac2a109836212566d562808b3847f71aa37c. Thanks